internal/oci/state.go

```diff
@@ -6,15 +6,25 @@ import (
 	"os"
 	"path/filepath"
 
-	"github.com/arm/remoteproc-runtime/internal/rootpath"
+	"github.com/arm/remoteproc-runtime/internal/userdirs"
 	"github.com/opencontainers/runtime-spec/specs-go"
 )
 
 const (
 	stateFileName = "state.json"
 )
 
-var stateDir = rootpath.Join("run", "remoteproc")
+var (
+	stateDir    string
+	stateDirErr error
+)
+
+func init() {
```

Comment:
I think we'd be better off avoiding `init()`. The error handling ergonomics (`panic`) are a bit nasty.

I'd much rather we had `getStateDir()` we'd call everywhere `stateDir` is used. To not pay the penalty of looking up the path, we could use `sync.Once`:

```go
var (
    stateDirOnce sync.Once 
	stateDir     string
	stateDirErr  error
)

func getStateDir() (string, error) {
	stateDirOnce.Do(func() {
		stateDir, stateDirErr = userdirs.RuntimeDir()
		if stateDirErr != nil {
			stateDirErr = fmt.Errorf("failed to get runtime directory: %w", stateDirErr)
			return
		}
	})
	return stateDir, stateDirErr
}
```